code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -128,13 +139,40 @@ def generate_on_your_data_llm_chain(
             "chat_history": chat_history,
         }
 
+    def generate_source_documents(self, search_results):
+        sources = []
+        for result in search_results:
+            source_url = self._find_all_http_or_https(result.get("source", ""))
+
+            metadata_dict = {
+                "id": result.get("id", ""),
+                "title": result.get("title", ""),
+                "source": source_url,
+                "chunk_id": result.get("chunk_id", ""),
+            }
+            sources.append(
+                Document(
+                    page_content=result["content"],
+                    metadata=metadata_dict,
+                )
+            )
+        return sources
+
+    def _find_all_http_or_https(self, original_source):
```

Comment:
Sorry I think I confused this by suggesting to extract a small part and then all of it! This is doing more than just the http/https part so maybe should be called something like 
```suggestion
    def _extract_source_url(self, original_source):
```